pkg/controllers/restore/controller.go

```diff
@@ -369,6 +370,10 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 		}
 
 		metadata := resourceData.Object[metadataMapKey].(map[string]interface{})
+		// ignore the secrets section in the service account to avoid missing token secret error in k8s 1.21
+		if resourceData.Object["kind"].(string) == "ServiceAccount" {
+			delete(currRestoreObj.Data.Object, secretsMapKey)
```

Comment:
Looking at func name/description `generateDependencyGraph` I would not expect it to "massage" objects. If there is already an existing function that does some sort of sanitize of objects to restore, I would add it there. If not, maybe worth introducing a separate func for this?

Reply:
Maybe worth logging this happened? At least at Debug level, if not Info.

Reply:
I put it here because the code below (within the same method) also modifies the object: correcting the `ownerRefs`. 

We should move both of them to a new method, but I'd like to leave it this way for now. 

A debug messaging is added right below this line.

Reply:
@jiaqiluo , thought about this as well, but modifying owner references is much more warranted in a function named `generateDependencyGraph` - one could expect owner references need massaging here; not so much with removing secret.
I guess you could argue secret is a dependency of an SA. I'm OK with not addressing it overall.